app/src/main/java/com/raghav/spacedawn/utils/AppApplication.kt

```diff
@@ -1,7 +1,25 @@
 package com.raghav.spacedawn.utils
 
 import android.app.Application
+import com.raghav.spacedawn.BuildConfig
 import dagger.hilt.android.HiltAndroidApp
 
 @HiltAndroidApp
-class AppApplication : Application()
+class AppApplication : Application() {
+
+    companion object {
+        lateinit var INSTANCE: AppApplication
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        INSTANCE = this
+    }
+
+    fun getLaunchLibraryBaseUrl(): String {
```

Comment:
can this logic not reside somewhere else?

Reply:
please you be more specific with the reviews? like
- problem with the existing code
- then it's solution

for eg. in this case
- this logic should not reside here because (reason)
- logic should be resided in example.kt (reason)

Reply:
- this logic should not reside here because (no relevance of the application class with something like this)
- logic should be resided in classes concerning themselves with these objects (repo or fragment)

Reply:
I am accessing this from my DI container , please tell an ideal solution and I'll implement the same